### PR TITLE
fix: not report error when encounter UnexpectedEof

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

refer #4376 

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

**Comment by BowenXiao1999** [comment]
Compared to simply ignore it, actually I'm more wondering why there is a Unexpected EoF.

**Comment by ZENOTME** [comment]
> Compared to simply ignore it, actually I'm more wondering why there is a Unexpected EoF.

We will maintain a 'Connection' with 'Client' and close the 'Connection' normally when 'Client' send the 'Terminate' message. I find the 'Client' doesn't send the 'Terminate' when it close. So the server still maintain the 'Connection' when the 'Client' has already exit. And the server try to read from this invalid 'Connection' and get the unexpectedEof.

So I think the problem is in 'Client' in tokio-postgres. It should implement the drop trait to send 'Terminate' message automatically. The weird thing is it implements a API to send 'Terminate' message but not implementing drop trait.
```
#[doc(hidden)]
    pub fn __private_api_close(&mut self) {
        self.inner.sender.close_channel()
    }
```

**Comment by BowenXiao1999** [comment]
> Compared to simply ignore it, actually I'm more wondering why there is a Unexpected EoF.
> 
> We will maintain a 'Connection' with 'Client' and close the 'Connection' normally when 'Client' send the 'Terminate' message. I find the 'Client' doesn't send the 'Terminate' when it close. So the server still maintain the 'Connection' when the 'Client' has already exit. And the server try to read from this invalid 'Connection' and get the unexpectedEof.
> 
> So I think the problem is in 'Client' in tokio-postgres. It should implement the drop trait to send 'Terminate' message automatically. The weird thing is it implements a API to send 'Terminate' message but not implementing drop trait.
> 
> ```
> #[doc(hidden)]
>     pub fn __private_api_close(&mut self) {
>         self.inner.sender.close_channel()
>     }
> ```

I have two findings:
1. It's not only relates to tokio::postgres. Psql also have this problem. But it's hard to classify the problem into psql.
2. Run one test file will have the ERROR log. But we running mutiple (`./e2e_test/batch/**/*.slt`) tests, seems like the ERROR log is gone

**Comment by ZENOTME** [comment]
> It's not only relates to tokio::postgres. Psql also have this problem. But it's hard to classify the problem into psql.
 
I try to run psql but it exit correctly.how to reproduce in psql?

**Comment by BowenXiao1999** [comment]
> > It's not only relates to tokio::postgres. Psql also have this problem. But it's hard to classify the problem into psql.
> 
> I try to run psql but it exit correctly.how to reproduce in psql?

Hmm just the procedure in #4376 . use risedev p + risdev slt. You sure it's good? 🤔

File: src/utils/pgwire/src/pg_protocol.rs [hunk 1]
```diff
@@ -103,7 +104,8 @@ where
         match self.do_process_inner().await {
             Ok(v) => v,
             Err(e) => {
-                tracing::error!("Error: {}", e);
+                let mut error_msg = String::new();
+                write!(&mut error_msg, "Error: {}", e).unwrap();
```
Comment (BowenXiao1999) [inline src/utils/pgwire/src/pg_protocol.rs:108]:
```suggestion
                // Execution error should not break current connection.
                // For unexpected eof, just break and not print to log.
                write!(&mut error_msg, "Error: {}", e).unwrap();
```

Comment (BowenXiao1999) [reply src/utils/pgwire/src/pg_protocol.rs:108]: Add some comments.

**Review by BowenXiao1999 — APPROVED** [review]
LGTM

**Pull request merged**